tapgarden/caretaker.go

```diff
@@ -1164,6 +1185,39 @@ func SortSeedlings(seedlings []*Seedling) []string {
 	return allSeedlings
 }
 
+// SortAssets sorts the batch assets such that assets that are group anchors are
+// partitioned from all other assets.
+func SortAssets(fullAssets []*asset.Asset,
+	anchorVerifier proof.GroupAnchorVerifier) ([]*asset.Asset,
+	[]*asset.Asset) {
+
+	var anchorAssets, nonAnchorAssets []*asset.Asset
```

Comment:
Since we expect the non-anchor witnesses to be the vast majority, should we go ahead and pre-initialize the slice to the size of `fullAssest`? Since if there are many thousands of assets, the time spent on growing the slice might not be insignificant.